### PR TITLE
telco5g: Export the "TARGET_RELEASE" var

### PR DESCRIPTION
**Pull request by liornoy**
This commit exports the "TARGET_RELEASE" var so the cnf-features-deploy cnf-tests "cnf-tests/hack/init-git-submodules.sh" script will consume it and perform checkout to the desired release on the git submodules.

This is important when testing non-master branch, e.g. release-4.15.

Also introduce a change that moves the call for "check_for_or" before running the init git submodules, 
to prevent overriding it.

Depends-On: https://github.com/openshift-kni/cnf-features-deploy/pull/1828

### CODE REVIEW

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by sshnaidm** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests periodic-ci-openshift-release-master-nightly-4.16-e2e-telco5g-cnftests

**Comment by sshnaidm** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

File: ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh [hunk 1]
```diff
@@ -293,14 +294,14 @@ fi
 
 pushd $CNF_REPO_DIR
 if [[ "$T5CI_VERSION" == "4.15" ]] || [[ "$T5CI_VERSION" == "4.16" ]]; then
+    check_for_pr "openshift-kni" "cnf-features-deploy"
```
Comment (danimjaffe) [inline ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh:297]: I think this needs to be outside of the if statement and not only for 4.15 and 4.16

File: ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh [hunk 2]
```diff
@@ -293,14 +294,14 @@ fi
...
     echo "Updating all submodules for >=4.15 versions"
     # git version 1.8 doesn't work well with forked repositories, requires a specific branch to be set
     sed -i "s@https://github.com/openshift/metallb-operator.git@https://github.com/openshift/metallb-operator.git\n        branch = main@" .gitmodules
-    git submodule update --init --force --recursive --remote
+    make init-git-submodules
     git submodule foreach --recursive 'echo $path `git config --get remote.origin.url` `git rev-parse HEAD`' | grep -v Entering > ${ARTIFACT_DIR}/hashes.txt || true
 fi
 echo "******** Checking out pull request for repository cnf-features-deploy if exists"
```
Comment (danimjaffe) [inline ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh:304]: And this needs to be moved as well

File: ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh (hunk 1 above)
Comment (liornoy) [reply ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh]: fixed thanks, we do want it to run on any other versions

File: ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh (hunk 2 above)
Comment (liornoy) [reply ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh]: You are right, thanks! fixed.

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-telco5g-cnftests

**Comment by liornoy** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-telco5g-cnftests

**Comment by sshnaidm** [comment]
/approve
/pj-rehearse ack
great job!

**Pull request merged**